Capital letters not allowed by osde2e

If you try and use a cluster name seed that has a capital letter in it the installation will fail. This error will be in the installation log:

```
2021/03/17 11:17:06 retryer.go:23: error during OCM attempt: api error: Cluster name 'scaleT-p7u-0001' isn't valid, it doesn't match regular expression '^[a-z]([-a-z0-9]*[a-z0-9])?$'
```

We should check for this prior to attempting to create the clusters

--- a/libs/common.py
+++ b/libs/common.py
@@ -3,6 +3,8 @@ import logging
 import time
 import os
 import errno
+import string
+import random
 
 _es_ignored_metadata = "before-suite-metrics,route-latencies,route-throughputs,route-availabilities,healthchecks,healthcheckIteration,status"
 
@@ -99,3 +101,17 @@ def _create_path(my_path):
         if e.errno != errno.EEXIST:
             logging.error(e)
             exit(1)
+
+def _generate_cluster_name_seed(cluster_name_seed):
+    _cluster_name_seed = cluster_name_seed
+    allowed_chars = string.ascii_lowercase + string.digits
+    for l in _cluster_name_seed:
+        if l not in allowed_chars:
+            logging.error('Cluster name seed is not valid: %s\nCluster name seed must contain only lowercase letters and digits.' % _cluster_name_seed)
+            exit(1)
+    random_string = ''.join(random.choice(allowed_chars) for j in range(3))
+    if len(_cluster_name_seed) > 6:
+        logging.warning('Cluster Name Seed too long (%d), truncated to %s' % (len(_cluster_name_seed), _cluster_name_seed[:6]))
+        _cluster_name_seed = _cluster_name_seed[:6]
+    _cluster_name_seed += "-" + random_string
+    return _cluster_name_seed

--- a/osde2e/osde2e-wrapper.py
+++ b/osde2e/osde2e-wrapper.py
@@ -23,8 +23,6 @@ import logging
 import git
 import threading
 import copy
-import string
-import random
 from libs import common
 from libs import parentParsers
 from ruamel.yaml import YAML
@@ -363,13 +361,7 @@ def main():
         logging.error('No ocm token supplied in configuration file: %s' % args.account_config)
         exit(1)
 
-    cluster_name_seed = args.cluster_name_seed
-    allowed_chars = string.ascii_lowercase + string.digits
-    random_string = ''.join(random.choice(allowed_chars) for j in range(3))
-    if len(cluster_name_seed) > 6:
-        logging.warning('Cluster Name Seed too long (%d), truncated to %s' % (len(cluster_name_seed), cluster_name_seed[:6]))
-        cluster_name_seed = cluster_name_seed[:6]
-    cluster_name_seed += "-" + random_string
+    cluster_name_seed = common._generate_cluster_name_seed(args.cluster_name_seed)
 
     # Set the user override if provided on the cli or generate a new one
     # if none is set

--- a/rosa/rosa-wrapper.py
+++ b/rosa/rosa-wrapper.py
@@ -23,8 +23,6 @@ import requests
 import urllib
 import logging
 import threading
-import random
-import string
 from libs import common
 from libs import parentParsers
 
@@ -345,6 +343,8 @@ def main():
         logging.error(err)
         exit(1)
 
+    cluster_name_seed = common._generate_cluster_name_seed(args.cluster_name_seed)
+
     rosa_cmnd = _verify_cmnd(args.rosa_cli,my_path)
 
     logging.info('Attempting to log in OCM using `rosa login`')
@@ -383,14 +383,6 @@ def main():
         else:
             logging.info('`rosa init` execution OK')
             logging.debug(rosa_init_stdout.strip().decode("utf-8"))
-
-    cluster_name_seed = args.cluster_name_seed
-    allowed_chars = string.ascii_lowercase + string.digits
-    random_string = ''.join(random.choice(allowed_chars) for j in range(3))
-    if len(cluster_name_seed) > 6:
-        logging.warning('Cluster Name Seed too long (%d), truncated to %s' % (len(cluster_name_seed), cluster_name_seed[:6]))
-        cluster_name_seed = cluster_name_seed[:6]
-    cluster_name_seed += "-" + random_string
 
     # launch watcher thread to report status
     logging.info('Launching watcher thread')

